sql/session.go

```diff
@@ -32,15 +32,20 @@ type Session interface {
 	GetAll() map[string]TypedValue
 	// ID returns the unique ID of the connection.
 	ID() uint32
+	// Warn stores the warning in the session.
+	Warn(warn *Warning)
+	// Warnings returns a copy of session warnings (from the most recent)
+	Warnings() []*Warning
 }
 
 // BaseSession is the basic session type.
 type BaseSession struct {
-	id     uint32
-	addr   string
-	user   string
-	mu     sync.RWMutex
-	config map[string]TypedValue
+	id       uint32
+	addr     string
+	user     string
+	mu       sync.RWMutex
+	config   map[string]TypedValue
+	warnings []*Warning
```

Comment:
when is this slice deleted?

Reply:
It can be deleted if we have some kind `Close` session.

Reply:
Isn't this supposed to be a per-query thing?

Reply:
Yep, should be deleted before new statement: https://mariadb.com/kb/en/library/show-warnings/

Reply:
Any better idea than extending Session interface by `ClearWarnings()`?

Reply:
we should add ClearWarnings() and call it before any query that is not a `show errors` or `show warnings`.

Reply:
OK, PTAL.